src/internal/m365/collection/site/collection.go

```diff
@@ -366,3 +358,26 @@ func (sc *Collection) handleListItems(
 	sc.stream <- item
 	progress <- struct{}{}
 }
+
+func finishPopulation(
```

Comment:
nit: Could you rename this to `updateStatus`, just to stay consistent with other services? I know the arg set has diverged, but still it helps while gleaning through the code.